source/common/protobuf/utility.h

```diff
@@ -42,6 +42,17 @@
        ? Protobuf::util::TimeUtil::DurationToSeconds((message).field_name())                       \
        : throw MissingFieldException(#field_name, (message)))
 
+// Convert an envoy::api::v2::core::Percent to a rounded integer or a default.
+// @param message supplies the proto message containing the field.
+// @param field_name supplies the field name in the message.
+// @param max_value supplies the maximum allowed integral value (e.g., 100, 10000, etc.).
+// @param default_value supplies the default if the field is not present.
+#define PROTOBUF_PERCENT_TO_ROUNDED_INTEGER_OR_DEFAULT(message, field_name, max_value,             \
+                                                       default_value)                              \
+  ((message).has_##field_name()                                                                    \
+       ? (static_cast<double>(max_value) * (message).field_name().value())                         \
```

Comment:
Are you missing a / 100?

If I'm trying to set a value 50% of 200, I'd expect the result to be 100, not 10,000.  Alternately if you're trying to avoid division maybe we can find a better name than percent?  But in that case I think the default value you're passing in is a default percent and needs the same multiplication.

I *think* with the current code the default of 50% and configured 50% will spit out different answers which seems like it can't be right.

Reply:
Agree this is a little confusing so here is what is happening:
1. "Percent" is the message we define in data-plane-api. It ranges from 0.0-1.0 (inclusive).
2. Here we are converting the percent to an integer in the range 0-max (inclusive).
3. "default" as specified is in the range 0-max (inclusive).

I can add more comments if that would help?

Reply:
Do we have more than one percent?  This is the one I was looking at.

message Percent {
  double value = 1 [(validate.rules).double = {gte: 0, lte: 100}];
}

Reply:
Oops, crap. Yes you are right. I think initially we had 0.0-1.0. I will fix!